source/crud/tests/README.rst

```diff
@@ -54,34 +42,56 @@ Each YAML file has the following keys:
 
   - ``description``: The name of the test.
 
-  - ``operation``: Document describing the operation to be executed. This will
-    have the following fields:
+  - ``skipReason`` (optional): If present, the test should be skipped and the
+    string value will specify a reason.
 
-    - ``name``: The name of the operation as defined in the specification. The
-      name `db-aggregate` refers to database-level aggregation.
+  - ``failPoint`` (optional): The ``configureFailPoint`` command document to run
+    to configure a fail point on the primary server.
 
-    - ``object``: The name of the object to perform the operation on. Can be
+  - ``clientOptions`` (optional): Names and values of options used to construct
+    the MongoClient for this test.
+
+  - ``operations``: Array of documents, each describing an operation to be
+    executed. Each document has the following fields:
+
+    - ``object`` (optional): The name of the object to perform the operation on. Can be
       "database" or "collection". Defaults to "collection" if undefined.
 
-    - ``arguments``: The names and values of arguments from the specification.
+    - ``collectionOptions`` (optional): Names and values of options used to
+      construct the collection object for this test.
 
-  - ``outcome``: Document describing the return value and/or expected state of
-    the collection after the operation is executed. This will have some or all
-    of the following fields:
+    - ``name``: The name of the operation as defined in the specification.
 
-    - ``error``: If ``true``, the test should expect an error or exception. Note
-      that some drivers may report server-side errors as a write error within a
-      write result object.
+    - ``arguments``: The names and values of arguments from the specification.
+
+    - ``error`` (optional): If ``true``, the test should expect the operation
+      to emit an error or exception. If ``false`` or omitted, drivers MUST
+      assert that no error occurred.
 
-    - ``result``: The return value from the operation. This will correspond to
-      an operation's result object as defined in the CRUD specification. This
-      field may be omitted if ``error`` is ``true``. If this field is present
-      and ``error`` is ``true`` (generally for multi-statement tests), the
-      result reports information about operations that succeeded before an
+    - ``result`` (optional): The result of executing the operation. This will
```

Comment:
@jmikola since `operations` is an array potentially containing multiple operations, we might need to clarify which 'result' the `result` key corresponds to? AFAICT we have 2 options:

1. Make this a list (say, `results`) with `N` entries where `N` is the number of elements in the `operations` list. Then the `n`-th entry in `results` is the expected result of the `n`-th operation in `operations`.
2. Restrict `result` to always refer to the return value of the *last* operation in the `operations` list.

Reply:
Each item in the "operations" list is its own "operation" which contains its own "result" field. For example: https://github.com/mongodb/specifications/blob/master/source/transactions/tests/delete.yml#L17-L32

Reply:
@prashantmital: I don't think there's a problem with the format as written. Does Shane's response clear this up?

The new format places `error` and `result` fields under each document element in the `operations` array. Previously in the legacy format, `error` and `result` appeared under `outcome` (parallel to the single `operation` field).

Reply:
Yes, my bad. Thanks @ShaneHarvey !